mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/blob/BlobStoreDeletedMessageVault.java

```diff
@@ -29,26 +32,36 @@
 import org.apache.james.task.Task;
 import org.apache.james.vault.DeletedMessage;
 import org.apache.james.vault.DeletedMessageVault;
+import org.apache.james.vault.RetentionConfiguration;
 import org.apache.james.vault.metadata.DeletedMessageMetadataVault;
 import org.apache.james.vault.metadata.DeletedMessageWithStorageInformation;
 import org.apache.james.vault.metadata.StorageInformation;
 import org.apache.james.vault.search.Query;
 import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Preconditions;
 
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
 public class BlobStoreDeletedMessageVault implements DeletedMessageVault {
+    private static final Logger LOGGER = LoggerFactory.getLogger(BlobStoreDeletedMessageVault.class);
+
     private final DeletedMessageMetadataVault messageMetadataVault;
     private final BlobStore blobStore;
     private final BucketNameGenerator nameGenerator;
+    private final Clock clock;
+    private final RetentionConfiguration retentionConfiguration;
 
-    public BlobStoreDeletedMessageVault(DeletedMessageMetadataVault messageMetadataVault, BlobStore blobStore, BucketNameGenerator nameGenerator) {
+    public BlobStoreDeletedMessageVault(DeletedMessageMetadataVault messageMetadataVault, BlobStore blobStore, BucketNameGenerator nameGenerator, Clock clock, RetentionConfiguration retentionConfiguration) {
```

Comment:
I'd like a builder for that (I'm fine for an other PR if you're in a hurry)

Reply:
I don't see the added value on what "should be" a package private constructor.

Reply:
so, make it a package private constructor

Reply:
https://github.com/linagora/james-project/pull/2521/commits/1f7fabad4d31a7db4e2b613c75b0c5030fb1b783 is answering this point (piggy back :-) )

Reply:
thank you